File: Livedata_Exam/Java/app/src/main/java/com/example/dgsw/livedata_exam/MainActivity.java
package com.example.dgsw.livedata_exam;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.room.Room;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText mTodoEditText = findViewById(R.id.todo_edit);
        TextView mResultTextView = findViewById(R.id.result_text);

        AppDatabase db = Room.databaseBuilder(this, AppDatabase.class, "todo-db-liveData1")
                .allowMainThreadQueries()
                .build();

        //관찰(Observe)도중 DB에 Insert시 UI 갱신
        db.todoDao().getAll().observe(this, todos -> mResultTextView.setText(todos.size() + "개"));

        //버튼 클릭시 DB에 Insert
        findViewById(R.id.add_button).setOnClickListener(v -> db.todoDao().insert(new Todo(mTodoEditText.getText().toString())));
    }
}

File: Kotlin/Livedata_Exam/app/src/main/java/com/example/dgsw/livedata_exam/MainActivity.kt
package com.example.dgsw.livedata_exam

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.room.Room

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val db = Room.databaseBuilder(
                applicationContext,
                AppDatabase::class.java,
                "todo-db-kt"
        ).allowMainThreadQueries().build()

        val editText = findViewById<EditText>(R.id.todo_edit)
        val textView = findViewById<TextView>(R.id.result_text)
        val button = findViewById<Button>(R.id.add_button)


        db.todoDao().getAll().observe(this, { todos ->
            textView.text = todos.toString()
        })

        button.setOnClickListener {
            db.todoDao().insert(Todo(editText.text.toString()))
        }

    }
}